src/main/haskell/kore/src/Kore/Builtin/List.hs

```diff
@@ -418,27 +444,70 @@ unify
       where
         asBuiltinDomainList = DV_ dvSort . BuiltinDomainList
 
-    unifyFramed
+    unifyFramedRight
+        :: (level ~ Object)
+        => Sort level
+        -> Seq p
+        -> SymbolOrAlias level
+        -> Seq p
+        -> p
+        -> m (expanded, proof)
+    unifyFramedRight resultSort list1 symConcat prefix2 frame2
+      | Seq.length prefix2 > Seq.length list1 =
+        return (ExpandedPattern.bottom, SimplificationProof)
+      | otherwise =
+        do
+            (prefixUnified, _) <- unifyConcrete resultSort prefix1 prefix2
+            (suffixUnified, _) <- simplifyChild frame2 listSuffix1
+            let result =
+                    (<$>)
```

Comment:
> Why not fmap if you're using it as prefix?

Because I hate that name. :laughing: As a practical matter, though, @u-- is working on [this](https://trello.com/c/BMUSsyAK) which will remove the application pattern entirely.

Reply:
`App_ symConcat <$>` would look just fine here.